src/managers/ThreadManager.js

```diff
@@ -0,0 +1,231 @@
+  constructor(channel, iterable) {
+    super(channel.client, iterable, ThreadChannel);
+
+    /**
+     * The channel this Manager belongs to
+     * @type {NewsChannel|TextChannel}
+     */
+    this.channel = channel;
+  }
+
+  /**
+   * The cache of this Manager
+   * @type {Collection<Snowflake, ThreadChannel>}
+   * @name ThreadManager#cache
+   */
+
+  add(thread) {
+    const existing = this.cache.get(thread.id);
+    if (existing) return existing;
+    this.cache.set(thread.id, thread);
+    return thread;
+  }
+
+  /**
+   * Data that can be resolved to give a Thread Channel object. This can be:
+   * * A ThreadChannel object
+   * * A Snowflake
+   * @typedef {ThreadChannel|Snowflake} ThreadChannelResolvable
+   */
+
+  /**
+   * Resolves a ThreadChannelResolvable to a Thread Channel object.
+   * @method resolve
+   * @memberof ThreadManager
+   * @instance
+   * @param {ThreadChannelResolvable} thread The ThreadChannel resolvable to resolve
+   * @returns {?ThreadChannel}
+   */
+
+  /**
+   * Resolves a ThreadChannelResolvable to a thread channel ID string.
+   * @method resolveID
+   * @memberof ThreadManager
+   * @instance
+   * @param {ThreadChannelResolvable} thread The ThreadChannel resolvable to resolve
+   * @returns {?Snowflake}
+   */
+
+  /**
+   * A number that is allowed to be the duration in minutes before a thread is automatically archived. This can be:
+   * * `60` (1 hour)
+   * * `1440` (1 day)
+   * * `4320` (3 days)
+   * * `10080` (7 days)
+   * @typedef {number} ThreadAutoArchiveDuration
+   */
+
+  /**
+   * Options for creating a thread
+   * @typedef {Object} ThreadCreateOptions
+   * @property {string} name The name of the new Thread
+   * @property {ThreadAutoArchiveDuration} autoArchiveDuration How long before the thread is automatically archived
+   * @property {MessageResolvable} [startMessage] The message to start a public or news thread from,
+   * creates a private thread if not provided
+   * @param {string} [reason] Reason for creating the thread
+   */
+
+  /**
+   * Creates a new thread in the channel.
+   * @param {ThreadCreateOptions} [options] Options
+   * @returns {Promise<ThreadChannel>}
+   * @example
+   * // Create a new public thread
+   * channel.threads
+   *   .create({
+   *     name: 'food-talk'
+   *     autoArchiveDuration: 60,
+   *     startMessage: channel.lastMessageID,
+   *     reason: 'Needed a separate thread for food',
+   *   })
+   *   .then(console.log)
+   *   .catch(console.error);
+   * // Create a new private thread
+   * channel.threads
+   *   .create({ name: 'mod-talk', autoArchiveDuration: 60, reason: 'Needed a separate thread for moderation' })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  async create({ name, autoArchiveDuration, startMessage, reason } = {}) {
+    let path = this.client.api.channels(this.channel.id);
+    if (startMessage) {
+      startMessage = this.channel.messages.resolveID(startMessage);
```

Comment:
This should probably ensure it's not null and throw if it is. Also, if you could avoid mutation that'd be sweet too!